### DISCUSSION TITLE
Different customers but using same plugin?

### DISCUSSION THREAD

**Discussion by jyu99** [fwupd/fwupd#2746] [General]
Our plugin need to support different pid,vid. Scenario as below  
We have ODM/OEM customer A, customer B ,etc.. Each ODM/OEM customer  pid & vid and could apply  unique LVFS account.
How to solve this problem ? Add customer pid and vid to our plugin quirks when we have new customer. or is any better way?

**Comment by hughsie** [comment]
Heu @jyu99. We already have this scenario now, e.g. with VIA chips. The LVFS account for each vendor has a list of vendor-ids it can update (e.g. USB:0x1234) and these can be the same if required.

> Add customer pid and vid to our plugin quirks

This is exactly what you need to do, and is completely fine to mix customers in the same quirk fiel.

**Comment by superm1** [comment]
Since it takes a while to get new quirks into distributions, my suggestion would be to populate the quirk with all the VID/PID that your plugin can support whether or not vendor has LVFS account.  That way vendor can create LVFS account and process for CAB asynchronously from fwupd release cadence.

Or if we have a way that we can _safely_ probe to determine it's a supportable device we can explore that instead of quirk.  Such as other USB descriptor information.

**Reply by jyu99** [reply]
But how to determine  the instance id ,if I want to polling all the HID peripheral? use the loosely condition ,e.g. `HIDRAW\` rather than `HIDRAW\VEN_093A&DEV_2801` 
`

**Reply by superm1** [reply]
You're probably right, there isn't a safe way to know unless it was all from Pixart VID. If it uses OEM or ODM VID then you need allow list.

**Reply by superm1** [reply]
Or if there is anything in USB descriptor even in OEM VID that is pixart specific as a hint?
